### PR TITLE
order-implementation

### PR DESCRIPTION
**Pull request by literandltx**

### CODE REVIEW

File: src/main/java/com/example/bookstore/repository/CartItemRepository.java [hunk 1]
```diff
@@ -2,7 +2,13 @@
...
 import com.example.bookstore.model.CartItem;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.transaction.annotation.Transactional;
 
 public interface CartItemRepository extends JpaRepository<CartItem, Long> {
-
+    @Modifying
+    @Transactional
```
Comment (Rommelua) [inline src/main/java/com/example/bookstore/repository/CartItemRepository.java:11]: use Transactional on servise layer

File: src/main/java/com/example/bookstore/repository/OrderRepository.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    @EntityGraph(attributePaths = {"orderItems", "orderItems.book"})
+    Page<Order> findOrdersByUser(User user, Pageable pageable);
+
+    @EntityGraph(attributePaths = "orderItems")
+    Optional<Order> findByIdAndUser(Long orderId, User user);
+
+    @Modifying
+    @Transactional
+    @Query("UPDATE Order o SET o.status = :status WHERE o.id = :orderId")
+    void updateOrderByStatus(Long orderId, Order.Status status);
```
Comment (Rommelua) [inline src/main/java/com/example/bookstore/repository/OrderRepository.java:24]: incorrect method name
Better delete it and use simple `update`

**Review by Rommelua — CHANGES_REQUESTED** [review]

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/bookstore/repository/OrderRepository.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    @EntityGraph(attributePaths = {"orderItems", "orderItems.book"})
+    Page<Order> findOrdersByUser(User user, Pageable pageable);
+
+    @EntityGraph(attributePaths = "orderItems")
+    Optional<Order> findByIdAndUser(Long orderId, User user);
+
+    @Modifying
+    @Transactional
+    @Query("UPDATE Order o SET o.status = :status WHERE o.id = :orderId")
+    void update(Long orderId, Order.Status status);
```
[the remark this answers is not in the document]
Comment (literandltx) [reply src/main/java/com/example/bookstore/repository/OrderRepository.java]: I don't quite understand what was meant, but there is no default update method, should I make my own custom one?

[the remark this answers is not in the document]
Comment (literandltx) [reply src/main/java/com/example/bookstore/repository/OrderRepository.java]: P.S for now made save() method

**Pull request merged**